pandas/core/ops/__init__.py

```diff
@@ -595,14 +595,25 @@ def _combine_series_frame(left, right, func, axis: int):
     right : Series
     func : binary operator
     axis : {0, 1}
+    str_rep : str
 
     Returns
     -------
     result : DataFrame
     """
     # We assume that self.align(other, ...) has already been called
     if axis == 0:
-        new_data = left._combine_match_index(right, func)
+        values = right._values
+        if isinstance(values, np.ndarray):
```

Comment:
Before long this will be moved into dispatch_to_series, which will gradually become de-facto dispatch_to_blocks.  At that point it may make sense to move it to another file.